cpp/unit/ability.h

```diff
@@ -42,17 +43,14 @@ using ability_set = std::bitset<16>;
 using ability_id_t = unsigned int;
```

Comment:
The value should be the size of of the enum class ability_type, not sure how to get the size of it here...